lang/LangPrimSource/PyrArrayPrimitives.cpp

```diff
@@ -2042,22 +2046,27 @@ int prArrayLace(struct VMGlobals *g, int numArgsPushed)
 				if (isKindOf(obj3, class_list)) {
 					obj3 = slotRawObject(&obj3->slots[0]); // get the list's array
 				}
-				if (obj3 && isKindOf(obj3, class_array)) {
+				if (obj3 && isKindOf(obj3, class_array) && obj3->size>0) {
 					m = j % obj3->size;
 					slotCopy(&obj2->slots[i],&obj3->slots[m]);
 				} else {
-					slotCopy(&obj2->slots[i],&slots[k]);
+					SetNil(&obj2->slots[i]);
 				}
 			} else {
 				slotCopy(&obj2->slots[i],&slots[k]);
 			}
 			k = (k+1) % obj1->size;
 			if (k == 0) j++;
 		}
+		obj2->size = n;
 	} else {
-		obj2 = instantiateObject(g->gc, obj1->classptr, n, true, true);
+		if (isKindOf(obj1, class_array)) {
+			obj2 = newPyrArray(g->gc, 0, 0, true);
+		} else {
+			obj2 = instantiateObject(g->gc, obj1->classptr, n, true, true);
```

Comment:
so this will mean that an empty list will return a new empty list?

Reply:
Correct. I changed that because I couldn't find a way to get `instantiateObject` to return an empty list instead of `[ nil ]`, even with `n = 0`.

Reply:
I see, makes sense. A good test would then be if the whole primitive works for subclasses of `List`, like `SortedList`.

Reply:
Pretty sure ```List``` and subclasses would get the object copy, as a ```List``` will fail the ```isKindOf``` test in the if clause on line 2063.

Reply:
The rest of the code seems to intend to treat `List` like arrays. Or is that my wrong impression?

Reply:
OK; checked myself, this works.

Reply:
There is one more thing, please check this again. If elements are Lists, the length calculation is still wrong:

```supercollider
[[1, 2, 3], [2, 3, 2], List[1, 2]].lace // [ 1, 2, 1 ]
[[1, 2, 3], [2, 3, 2], List[1, 2].asArray].lace // [ 1, 2, 1, 2, 3, 2 ]
[[1, 2, 3], [2, 3, 2], List[1, 2]].lace(5) // [ 1, 2, 1, 2, 3 ]
```

sorry, all the rest looks good so far. Maybe you can add a few tests that mix `List` and `Array`?

Reply:
Done. Note that this is adding new functionality, but agree that it makes sense to add.